File: tests/transpiler/x/java/inner_join.java
public class Main {
    static Data1[] customers = new Data1[]{new Data1(1, "Alice"), new Data1(2, "Bob"), new Data1(3, "Charlie")};
    static class Data1 {
        int id;
        String name;
        Data1(int id, String name) {
            this.id = id;
            this.name = name;
        }
    }

    static Data2[] orders = new Data2[]{new Data2(100, 1, 250), new Data2(101, 2, 125), new Data2(102, 1, 300), new Data2(103, 4, 80)};
    static class Data2 {
        int id;
        int customerId;
        int total;
        Data2(int id, int customerId, int total) {
            this.id = id;
            this.customerId = customerId;
            this.total = total;
        }
    }

    static java.util.List<Result4> result = new java.util.ArrayList<Result4>() {{ for (var o : orders) { for (var c : customers) { if (o.customerId == c.id) { add(new Result4(o.id, c.name, o.total)); } } }}};
    static class Data3 {
         orderId;
         customerName;
         total;
        Data3( orderId,  customerName,  total) {
            this.orderId = orderId;
            this.customerName = customerName;
            this.total = total;
        }
    }

    static class Result4 {
         orderId;
         customerName;
         total;
        Result4( orderId,  customerName,  total) {
            this.orderId = orderId;
            this.customerName = customerName;
            this.total = total;
        }
    }


    public static void main(String[] args) {
        System.out.println("--- Orders with customer info ---");
        for (var entry : result) {
            System.out.println("Order" + " " + entry.orderId + " " + "by" + " " + entry.customerName + " " + "- $" + " " + entry.total);
        }
    }
}

File: tests/transpiler/x/fs/inner_join.fs
// Generated 2025-07-21 15:37 +0700
open System

type Anon1 = {
    mutable id: int
    mutable name: string
}
type Anon2 = {
    mutable id: int
    mutable name: string
}
type Anon3 = {
    mutable id: int
    mutable customerId: int
    mutable total: int
}
type Anon4 = {
    mutable id: int
    mutable customerId: int
    mutable total: int
}
type Anon5 = {
    mutable orderId: obj
    mutable customerName: obj
    mutable total: obj
}
type Anon6 = {
    mutable orderId: obj
    mutable customerName: obj
    mutable total: obj
}
let customers: Anon2 list = [{ id = 1; name = "Alice" }; { id = 2; name = "Bob" }; { id = 3; name = "Charlie" }]
let orders: Anon4 list = [{ id = 100; customerId = 1; total = 250 }; { id = 101; customerId = 2; total = 125 }; { id = 102; customerId = 1; total = 300 }; { id = 103; customerId = 4; total = 80 }]
let result: Anon6 list = [ for o in orders do for c in customers do if (o.customerId) = (c.id) then yield { orderId = o.id; customerName = c.name; total = o.total } ]
printfn "%s" (string "--- Orders with customer info ---")
for entry in result do
printfn "%s" (String.concat " " [string "Order"; string (entry.orderId); string "by"; string (entry.customerName); string "- $"; string (entry.total)])
